pkg/logql/log/labels.go

```diff
@@ -437,6 +438,32 @@ func (b *LabelsBuilder) UnsortedLabels(buf labels.Labels, categories ...LabelCat
 	return buf
 }
 
+var stringMapPool = sync.Pool{
```

Comment:
Would it make sense to wrap this in a specific type so we don't have to typcast when we decide to use it?

Reply:
funnily enough, I did exactly this already but forgot to push it